content/Phoenix for Rails developers - Bruno Vezoli.md

```diff
@@ -0,0 +1,347 @@
+
+In time, this led to an independent web framework but with that special flavor we all love from Rails. There’s one difference though, since Elixir was built on top of Erlang, which was designed to build massively scalable applications, all apps built with Phoenix inherit this perk. This means that you can code away without having to worry how your app will perform once you reach the big leagues. This is something with which many Rails apps struggle and it’s the main reason why Jose Valim developed Elixir in the first place.
+
+But enough preaching, let’s see some code!
+
+
+### Comparison by example
+
+This post does not intend to be a complete guide on Phoenix, it’s a big framework with lots of things to get into. Instead, I’ll show a side by side comparison of how things are done in both frameworks using the scaffolding they provide to throw some insight on their best practices with a classic blog example.
+
+The idea is to show how similar they are and get you hyped enough to try it by yourself.
+
+### File structure
+
+![File structure comparison](images/phoenix-rails-project-structure.png)
+
+No opinionated framwework would be complete without a take on how files should be organized. Phoenix's structure inherits from how packages (Elixir's equivalent for gems) are structured, all custom codes goes inside the `lib/` folder. Inside it you'll find right away the main difference between both frameworks in this aspect: Phoenix separates the layer of web presentation from the business logic. In this example we have a `blog/` folder, which cointains the basic modules to build our business logic, and the `blog_web` which looks pretty similar to the Rails' `app/` directory. There are some subtle differences though but we'll take a deeper look into the difference in the following sections.
+
+### Scaffold away
+
+As I mentioned, both frameworks provide commands to scaffold different parts of the application. For this example we'll compare the output of generating a full resource including views and routes, but first the commands used:
+
+```bash
+bin/rails generate scaffold Article title:string text:text
+```
+
+```bash
+mix phx.gen.html Articles Article articles title:string text:text
+```
+
+Once more they look alike with subtle differences, in this case Phoenix goes for a more verbose fashion by asking us to specify the context module where it'll be created, the name of the schema module (equivalent to the model) and it's plural to be used as the table name.
```

Comment:
`and it's plural to be used as the table name.` The possessive `its` doesn't have an apostrophe.